src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeValueConverter.cs

```diff
@@ -23,19 +23,21 @@ public override InputEnumTypeValue Read(ref Utf8JsonReader reader, Type typeToCo
         public override void Write(Utf8JsonWriter writer, InputEnumTypeValue value, JsonSerializerOptions options)
             => throw new NotSupportedException("Writing not supported");
 
-        public static InputEnumTypeValue CreateEnumTypeValue(ref Utf8JsonReader reader, string? id, string? name, JsonSerializerOptions options, ReferenceResolver resolver)
+        private static InputEnumTypeValue CreateEnumTypeValue(ref Utf8JsonReader reader, string? id, string? name, JsonSerializerOptions options, ReferenceResolver resolver)
         {
             var isFirstProperty = id == null;
-            object? value = null;
+            JsonElement? rawValue = null;
+            InputPrimitiveType? valueType = null;
             string? description = null;
             IReadOnlyList<InputDecoratorInfo>? decorators = null;
             while (reader.TokenType != JsonTokenType.EndObject)
             {
                 var isKnownProperty = reader.TryReadReferenceId(ref isFirstProperty, ref id)
-                    || reader.TryReadString(nameof(InputEnumTypeValue.Name), ref name)
-                    || reader.TryReadEnumValue(nameof(InputEnumTypeValue.Value), ref value)
-                    || reader.TryReadString(nameof(InputEnumTypeValue.Description), ref description)
-                    || reader.TryReadWithConverter(nameof(InputEnumTypeValue.Decorators), options, ref decorators);
+                    || reader.TryReadString("name", ref name)
+                    || reader.TryReadWithConverter("value", options, ref rawValue)
+                    || reader.TryReadWithConverter("valueType", options, ref valueType)
```

Comment:
is this valueType used anywhere else apart from this deserialization? If not, do we need it in the InputEnumTypeValue?

Reply:
we used it below to determine which concrete type we want to use for the raw value (whose type is `object`)

Before this PR, we are using the `ValueType` from the enum therefore there is a lot of workarounds, which are all removed now